fili-core/src/main/java/com/yahoo/bard/webservice/logging/blocks/BardQueryInfo.java

```diff
@@ -45,48 +37,56 @@
      */
     public BardQueryInfo(String queryType) {
         this.type = queryType;
+        this.queryCounter = Stream.of(
+                new AbstractMap.SimpleImmutableEntry<>(WEIGHT_CHECK, new AtomicInteger()),
+                new AbstractMap.SimpleImmutableEntry<>(FACT_QUERIES, new AtomicInteger()),
+                new AbstractMap.SimpleImmutableEntry<>(FACT_QUERY_CACHE_HIT, new AtomicInteger())
+        ).collect(Collectors.toMap(
+                AbstractMap.SimpleImmutableEntry::getKey,
+                AbstractMap.SimpleImmutableEntry::getValue
+        ));
+    }
+
+    /**
+     * Retrieves {@link com.yahoo.bard.webservice.logging.blocks.BardQueryInfo} from
+     * {@link com.yahoo.bard.webservice.logging.RequestLog}.
+     *
+     * @return {@link com.yahoo.bard.webservice.logging.blocks.BardQueryInfo} from
+     * {@link com.yahoo.bard.webservice.logging.RequestLog}
+     */
+    public static BardQueryInfo getBardQueryInfo() {
+        return ((BardQueryInfo) RequestLog.retrieve(BardQueryInfo.class));
```

Comment:
The `protected static BardQueryInfo getBardQueryInfo()` below is moved up to here(since static method goes first)